File: planet-examples/src/main/java/org/saar/minecraft/generator/FlatTerrainGenerator.java
package org.saar.minecraft.generator;

import org.saar.minecraft.Blocks;
import org.saar.minecraft.Chunk;

public class FlatTerrainGenerator implements WorldGenerator {

    private final int height;

    public FlatTerrainGenerator(int height) {
        this.height = height;
    }

    @Override
    public void generateChunk(Chunk chunk) {
        for (int x = 0; x < 16; x++) {
            for (int z = 0; z < 16; z++) {
                for (int y = 0; y < this.height; y++) {
                    chunk.setBlock(x, y, z, Blocks.DIRT);
                }
                chunk.setBlock(x, this.height, z, Blocks.GRASS);
            }
        }
    }
}

File: planet-examples/src/main/java/org/saar/minecraft/generator/FlatTerrainGenerator.kt
package org.saar.minecraft.generator

import org.saar.minecraft.Blocks
import org.saar.minecraft.Chunk

class FlatTerrainGenerator(private val height: Int) : WorldGenerator {
    override fun generateChunk(chunk: Chunk) {
        for (x in 0..15) {
            for (z in 0..15) {
                for (y in 0..<this.height) {
                    chunk.setBlock(x, y, z, Blocks.DIRT)
                }
                chunk.setBlock(x, this.height, z, Blocks.GRASS)
            }
        }
    }
}
